### ISSUE TITLE
Soap Headers not generated correctly

### ISSUE THREAD

**Issue by banderson5144** [pepri/wizdler#10]
Using the Salesforce Partner wsdl, the Soap headers are not being generated correctly. For example making the query call should use these headers:

<operation name="query">
<soap:operation soapAction=""/>
<input>
<soap:header use="literal" message="tns:Header" part="SessionHeader"/>
<soap:header use="literal" message="tns:Header" part="CallOptions"/>
<soap:header use="literal" message="tns:Header" part="QueryOptions"/>
<soap:header use="literal" message="tns:Header" part="MruHeader"/>
<soap:header use="literal" message="tns:Header" part="PackageVersionHeader"/>

However, this header is repeated each time:

<element name="LoginScopeHeader">
<complexType>
<sequence>
<element name="organizationId" type="tns:ID"/>
<element name="portalId" type="tns:ID" minOccurs="0"/>
</sequence>
</complexType>
</element

Generated SOAP Message:

<Envelope xmlns="http://schemas.xmlsoap.org/soap/envelope/">
    <Header>
        <LoginScopeHeader xmlns="urn:partner.soap.sforce.com">
            <organizationId>[string]</organizationId>
            <portalId>[string?]</portalId>
        </LoginScopeHeader>
        <LoginScopeHeader xmlns="urn:partner.soap.sforce.com">
            <organizationId>[string]</organizationId>
            <portalId>[string?]</portalId>
        </LoginScopeHeader>
        <LoginScopeHeader xmlns="urn:partner.soap.sforce.com">
            <organizationId>[string]</organizationId>
            <portalId>[string?]</portalId>
        </LoginScopeHeader>
        <LoginScopeHeader xmlns="urn:partner.soap.sforce.com">
            <organizationId>[string]</organizationId>
            <portalId>[string?]</portalId>
        </LoginScopeHeader>
        <LoginScopeHeader xmlns="urn:partner.soap.sforce.com">
            <organizationId>[string]</organizationId>
            <portalId>[string?]</portalId>
        </LoginScopeHeader>
    </Header>
    <Body>
        <query xmlns="urn:partner.soap.sforce.com">
            <queryString>[string]</queryString>
        </query>
    </Body>
</Envelope>

**Comment by banderson5144** [comment]
Using the Salesforce Partner wsdl, the Soap headers are not being generated correctly. For example making the query call should use these headers:

&lt;operation name=&quot;query&quot;&gt;
&lt;soap:operation soapAction=&quot;&quot;/&gt;
&lt;input&gt;
&lt;soap:header use=&quot;literal&quot; message=&quot;tns:Header&quot; part=&quot;SessionHeader&quot;/&gt;
&lt;soap:header use=&quot;literal&quot; message=&quot;tns:Header&quot; part=&quot;CallOptions&quot;/&gt;
&lt;soap:header use=&quot;literal&quot; message=&quot;tns:Header&quot; part=&quot;QueryOptions&quot;/&gt;
&lt;soap:header use=&quot;literal&quot; message=&quot;tns:Header&quot; part=&quot;MruHeader&quot;/&gt;
&lt;soap:header use=&quot;literal&quot; message=&quot;tns:Header&quot; part=&quot;PackageVersionHeader&quot;/&gt;

However, this header is repeated each time:

&lt;element name=&quot;LoginScopeHeader&quot;&gt;
&lt;complexType&gt;
&lt;sequence&gt;
&lt;element name=&quot;organizationId&quot; type=&quot;tns:ID&quot;/&gt;
&lt;element name=&quot;portalId&quot; type=&quot;tns:ID&quot; minOccurs=&quot;0&quot;/&gt;
&lt;/sequence&gt;
&lt;/complexType&gt;
&lt;/element&gt;

Generated SOAP Message:

&lt;Envelope xmlns=&quot;http://schemas.xmlsoap.org/soap/envelope/&quot;&gt;
    &lt;Header&gt;
        &lt;LoginScopeHeader xmlns=&quot;urn:partner.soap.sforce.com&quot;&gt;
            &lt;organizationId&gt;[string]&lt;/organizationId&gt;
            &lt;portalId&gt;[string?]&lt;/portalId&gt;
        &lt;/LoginScopeHeader&gt;
        &lt;LoginScopeHeader xmlns=&quot;urn:partner.soap.sforce.com&quot;&gt;
            &lt;organizationId&gt;[string]&lt;/organizationId&gt;
            &lt;portalId&gt;[string?]&lt;/portalId&gt;
        &lt;/LoginScopeHeader&gt;
        &lt;LoginScopeHeader xmlns=&quot;urn:partner.soap.sforce.com&quot;&gt;
            &lt;organizationId&gt;[string]&lt;/organizationId&gt;
            &lt;portalId&gt;[string?]&lt;/portalId&gt;
        &lt;/LoginScopeHeader&gt;
        &lt;LoginScopeHeader xmlns=&quot;urn:partner.soap.sforce.com&quot;&gt;
            &lt;organizationId&gt;[string]&lt;/organizationId&gt;
            &lt;portalId&gt;[string?]&lt;/portalId&gt;
        &lt;/LoginScopeHeader&gt;
        &lt;LoginScopeHeader xmlns=&quot;urn:partner.soap.sforce.com&quot;&gt;
            &lt;organizationId&gt;[string]&lt;/organizationId&gt;
            &lt;portalId&gt;[string?]&lt;/portalId&gt;
        &lt;/LoginScopeHeader&gt;
    &lt;/Header&gt;
    &lt;Body&gt;
        &lt;query xmlns=&quot;urn:partner.soap.sforce.com&quot;&gt;
            &lt;queryString&gt;[string]&lt;/queryString&gt;
        &lt;/query&gt;
    &lt;/Body&gt;
&lt;/Envelope&gt;

**Comment by banderson5144** [comment]
I made a fix locally to my extensions to the WSDL.js file

``` javascript
Wsdl._generateSoapMessage = function(ctx, soapVersion) {
    var ns = Wsdl.ns[soapVersion + 'Env'];
    var doc = document.implementation.createDocument(ns, 'Envelope', null);
    // SOAP headers
    var headers = ctx.operation.input[soapVersion].headers;
    if (headers.length) {
        var hdr = doc.createElementNS(ns, 'Header');
        doc.documentElement.appendChild(hdr);
        for (var i = 0, n = headers.length; i < n; ++i) {
            var message = ctx.wsdl.messages[headers[i].message.full];
            var theEl;

            for(var j = 0; j < message.parts.length; j++)
            {
                if(headers[i].parts == message.parts[j].name)
                {
                    theEl = message.parts[j].element;
                    break;
                }
            }

            var el = ctx.wsdl.generator.writeXml(theEl).documentElement;
            hdr.appendChild(doc.importNode(el));
        }
    }
```

**Comment by pepri** [comment]
Thank you for your fix. I merged it into version 1.15, which is now available on the web store.
